packages/primitives/radio/src/Root.native.tsx

```diff
@@ -1,73 +1,73 @@
 import React from 'react'
 import { TouchableWithoutFeedback as Touchable, View } from 'react-native'
-import { component, startWithType, mapContext, mapDefaultProps } from 'refun'
+import { component, startWithType, mapContext, mapDefaultProps, mapHandlers } from 'refun'
 import { normalizeNativeStyle } from 'stili'
 import { TRadioInput } from './types'
 import { RadioContext } from './context'
 
-const styles = normalizeNativeStyle({
+const visibleWrapperStyles = normalizeNativeStyle({
   flexGrow: 1,
   flexShrink: 1,
   alignSelf: 'stretch',
   minWidth: 0,
-  justifyContent: 'space-between',
+  width: 10,
+  height: 10,
+  borderColor: '#000',
+  borderStyle: 'solid',
+  borderWidth: 1,
+  borderRadius: 50,
+  justifyContent: 'center',
+  alignItems: 'center',
 })
 
+const visibleBullet = normalizeNativeStyle({
+  width: 2,
+  height: 2,
+  backgroundColor: '#000',
+  borderRadius: 50,
+  flexGrow: 0,
+})
+
+const VisibleRadio = ({ isChecked }: { isChecked: boolean}) => (
+  <View style={visibleWrapperStyles}>
+    {isChecked && <View style={visibleBullet}/>}
+  </View>
+)
+
 export const RadioInput = component(
   startWithType<TRadioInput>(),
+  mapContext(RadioContext),
+  mapHandlers({
+    onPress: ({ isDisabled, setGroupValue, value }) => () => (isDisabled ? undefined : setGroupValue(value)),
+  }),
   mapDefaultProps({
     isDisabled: false,
-  }),
-  mapContext(RadioContext)
+  })
 )(({
+  children,
+  onPress,
+  isVisible,
   groupName,
-  id,
   accessibilityLabel,
-  isDisabled,
-  onChange,
+  id,
   value,
   groupValue,
-  setGroupValue,
-}) => (
-  <Touchable
-    style={styles}
-    testID={`${groupName}-${id}`}
-    accessibilityLabel={accessibilityLabel}
-    onPress={(evt) => {
-      if (isDisabled) {
-        return false
-      }
+}) => {
+  const isChecked = groupValue === value
 
-      setGroupValue(value)
-
-      if (typeof onChange === 'function') {
-        onChange(evt)
-      }
-    }}
-  >
-    <View style={{
-      width: 10,
-      height: 10,
-      borderColor: '#000',
-      borderStyle: 'solid',
-      borderWidth: 1,
-      borderRadius: 50,
-      justifyContent: 'center',
-      alignItems: 'center',
-    }}
+  return (
+    <Touchable
+      testID={`${groupName}-${id}`}
```

Comment:
* Making same string every frame. `id` is unique enough. No need to add it to `groupName`.
* Also different with Web `id`, which is just `id`.

Reply:
in web it's just `id`, that's why they're different.
But I can make them match for now and we see if we need something more specific in the future